ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -49,18 +48,22 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+// CHECKSTYLE_RULES.OFF: ClassDataAbstractionCoupling
 public final class CommandParser {
+  // CHECKSTYLE_RULES.ON: ClassDataAbstractionCoupling
   private static final String QUOTED_STRING_OR_WHITESPACE = "('([^']*|(''))*')|\\s+";
-  private static final Pattern SET_PROPERTY =
-      Pattern.compile("\\s*SET\\s+'((?:[^']*|(?:''))*)'\\s*=\\s*'((?:[^']*|(?:''))*)'\\s*;\\s*");
-  private static final Pattern UNSET_PROPERTY =
-      Pattern.compile("\\s*UNSET\\s+'((?:[^']*|(?:''))*)'\\s*;\\s*");
+  private static final Pattern SET_PROPERTY = Pattern.compile(
+      "\\s*SET\\s+'((?:[^']*|(?:''))*)'\\s*=\\s*'((?:[^']*|(?:''))*)'\\s*;\\s*",
+      Pattern.CASE_INSENSITIVE);
+  private static final Pattern UNSET_PROPERTY = Pattern.compile(
+      "\\s*UNSET\\s+'((?:[^']*|(?:''))*)'\\s*;\\s*", Pattern.CASE_INSENSITIVE);
   private static final Pattern DROP_CONNECTOR =
-      Pattern.compile("\\s*DROP\\s+CONNECTOR\\s+(.*?)\\s*;\\s*");
+      Pattern.compile("\\s*DROP\\s+CONNECTOR\\s+(.*?)\\s*;\\s*", Pattern.CASE_INSENSITIVE);
   private static final Pattern DEFINE_VARIABLE = Pattern.compile(
-      "\\s*DEFINE\\s+([a-zA-Z_][a-zA-Z0-9_@]*)\\s*=\\s*'((?:[^']*|(?:''))*)'\\s*;\\s*");
-  private static final Pattern UNDEFINE_VARIABLE =
-      Pattern.compile("\\s*UNDEFINE\\s+([a-zA-Z_][a-zA-Z0-9_@]*)\\s*;\\s*");
+      "\\s*DEFINE\\s+([a-zA-Z_][a-zA-Z0-9_@]*)\\s*=\\s*'((?:[^']*|(?:''))*)'\\s*;\\s*",
+      Pattern.CASE_INSENSITIVE);
+  private static final Pattern UNDEFINE_VARIABLE = Pattern.compile(
+      "\\s*UNDEFINE\\s+([a-zA-Z_][a-zA-Z0-9_@]*)\\s*;\\s*", Pattern.CASE_INSENSITIVE);
```

Comment:
Do we still need the `toUpperCase()` call in `transformToSqlCommand()` now that we have case-insensitivity?

Reply:
Yup, it's used to determine the type of command

Reply:
But why do we need to uppercase the command if the regex is case-insensitive now?

Reply:
The regexes are used to determine things like the name of a connector being dropped. There is some logic before that for determining the type of command or if the command is supported, which works by finding specific keywords in the sql string.

Reply:
Ah, got it. Would it be equivalent if we removed the `toUpperCase()` and instead replaced the `.equals()` for token comparison with `.equalsIgnoreCase()` instead? It's mildly less efficient but I think it improves readability by not separating the logic for case-insensitive comparison across two different places.

Reply:
There are some places where it uses `contains` instead of `equals`, for example `tokens.contains(VALUES)`, so removing the `toUpperCase` would actually make it more confusing. I've added some comments to clarify what `tokens` is supposed to be.

Reply:
Sounds good. Thanks!